### PR TITLE
feat(app): Display run setup task completion

### PR DESCRIPTION
**Pull request by sfoster1**
In the run page on both app and ODD, you get an indication of completion when you get all your instruments and modules and deck stuff present and calibrated. But those are just two of the steps presented equally in the run setup page, and the rest of the steps don't get anything similar. It leads people to wonder whether they've set things up properly.

This PR adds similar styling and completion semantics for the other tasks in the run setup screen to fix this issue. Specifically, LPC gets a "confirm offsets" button (which will confirm offsets even if you haven't run LPC - makes it more apparent that that's a separate option) and labware and liquids get generic confirm buttons.

There's also a couple other visual fixes:
- On desktop, the "back to top" button in run setup is now where figma thinks it is, outside the run-setup content area. This allows some refactoring of component props
- On desktop, there was an issue with the react-router upgrade (I think - it's also in the latest IR alpha) that means that if you had an ongoing run, you couldn't view anything but run details without getting instantly navigated back to run details

This implements this figma: https://www.figma.com/design/Rwdt9R0aERFC55oTLDTlqY/8.0-September-Release-File?node-id=39-35830&t=l6vwJjQsfyVeovfC-4

## To come out of draft
- [x] implement for ODD
- [x] rebase onto release
- [x] "are you sure" modal on desktop
- [x] "are you sure" modal on ODD

## Review requests
- This is some pretty complex UI - do you agree with how I've done this?
- Some of this is pretty ugly, in large part because this is old code that I'm cleaning up. There's some duplicated logic in the run details and some pretty ugly typing. What I'd like to do is merge this since it implements some features nicely and then follow up with a refactor to get the size of some of these files down and enforce nicer separation between everything.

## Testing
- [x] Desktop green checks on flex
- [x] Desktop green checks on OT-2 (yes, this has to be different because the steps can be different here)

Closes RSQ-7

### CODE REVIEW

File: app/src/organisms/Devices/ProtocolRun/ConfirmMissingStepsModal.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      type="warning"
+      onClose={onCloseClick}
+    >
+      <Flex flexDirection={DIRECTION_COLUMN} fontSize={TYPOGRAPHY.fontSizeP}>
+        <LegacyStyledText paddingBottom={SPACING.spacing4}>
+          {t('you_havent_confirmed', {
+            missingSteps: new Intl.ListFormat('en', {
+              style: 'short',
+              type: 'conjunction',
+            }).format(missingSteps.map(step => t(step))),
```
Comment (mjhuff) [inline app/src/organisms/Devices/ProtocolRun/ConfirmMissingStepsModal.tsx:45]: Oh, this is slick.

File: app/src/pages/ProtocolSetup/index.tsx [hunk 2]
```diff
@@ -872,6 +940,23 @@ export function ProtocolSetup(): JSX.Element {
...
   >([])
+  const [labwareConfirmed, setLabwareConfirmed] = React.useState<boolean>(false)
+  const [liquidsConfirmed, setLiquidsConfirmed] = React.useState<boolean>(false)
+  const [offsetsConfirmed, setOffsetsConfirmed] = React.useState<boolean>(false)
+  const missingSteps = [
+    !offsetsConfirmed ? t('applied_labware_offsets') : null,
+    !labwareConfirmed ? t('labware_placement') : null,
+    !liquidsConfirmed ? t('liquids') : null,
+  ].filter(s => s != null)
+  console.log(missingSteps)
```
Comment (mjhuff) [inline app/src/pages/ProtocolSetup/index.tsx]: stray log?

File: app/src/organisms/ProtocolSetupOffsets/index.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+      makeSnackbar(lpcDisabledReason)
+    }
+  }
+
+  const labwareDefinitions = getLabwareDefinitionsFromCommands(
+    mostRecentAnalysis?.commands ?? []
+  )
+  const { data: runRecord } = useNotifyRunQuery(runId, { staleTime: Infinity })
+  const currentOffsets = runRecord?.data?.labwareOffsets ?? []
+  const sortedOffsets: LabwareOffset[] =
```
Comment (mjhuff) [inline app/src/organisms/ProtocolSetupOffsets/index.tsx:54]: Yeah, it feels like this shouldn't be the best way to do this, but I'm pretty sure you're right.

File: app/src/pages/Devices/ProtocolRunDetails/index.tsx [hunk 4]
```diff
@@ -193,37 +202,69 @@ function PageContents(props: PageContentsProps): JSX.Element {
...
+        />
+      ),
+      backToTop: null,
+    },
   }
-
-  const protocolRunDetailsContent = protocolRunDetailsContentByTab[
-    protocolRunDetailsTab
-  ] ?? (
+  console.log(`run details tab is ${protocolRunDetailsTab}`)
```
Comment (mjhuff) [inline app/src/pages/Devices/ProtocolRunDetails/index.tsx]: stray log?

File: app/src/pages/ProtocolSetup/index.tsx [hunk 5]
```diff
@@ -883,9 +968,13 @@ export function ProtocolSetup(): JSX.Element {
         runId={runId}
         setSetupScreen={setSetupScreen}
         confirmAttachment={confirmAttachment}
+        confirmStepsComplete={confirmMissingSteps}
```
Comment (mjhuff) [inline app/src/pages/ProtocolSetup/index.tsx:971]: Definitely not a now thing or a thing you've introduced, but this `setupComponentByScreen` should at the least be in a useMemo but more likely, it's own component, so not every component is instantiated on each render.

**Review by mjhuff — APPROVED** [review]
Nice, nice. Agreed that we should follow a lot of this up with a refactor at some point. The UI is complex (which seems unavoidable here, at least without thinking about this much more), but I can follow it. Thank you for doing this!

File: app/src/organisms/ProtocolSetupOffsets/index.tsx (hunk 3 above)
Comment (sfoster1) [reply app/src/organisms/ProtocolSetupOffsets/index.tsx:54]: It should absolutely be a reusable hook at some point

**Pull request merged**